### PR TITLE
fix cli test command for agent directory

### PR DESCRIPTION
**Pull request by solarw**
## Proposed changes

fix cli test command for agent directory

## Fixes

https://github.com/valory-xyz/open-aea/issues/547

https://github.com/valory-xyz/open-aea/issues/551

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes and CI passes too
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

DELETE INCLUSIVE THIS AND BELOW FOR STANDARD PR
------

## Release summary

Version number: [e.g. 1.0.1]

## Release details

Describe in short the main changes with the new release.

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side), from `develop`
- [ ] Lint and unit tests pass locally and in CI
- [ ] I have checked the fingerprint hashes are correct by running (`aea packages lock --check`)
- [ ] I have regenerated the latest API docs
- [ ] I built the documentation and updated it with the latest changes
- [ ] I have added an item in `HISTORY.md` for this release
- [ ] I bumped the version number in the `aea/__version__.py` file.
- [ ] I bumped the version number in every Docker image of the repo and published it. Also, I built and published them with tag `latest`  
      (check the READMEs of [`aea-develop`](../develop-image/README.md#publish) 
      and [`aea-user`](../user-image/README.md#publish))
- [ ] I have pushed the latest packages to the registry.
- [ ] I have uploaded the latest `aea` to PyPI.
- [ ] I have uploaded the latest plugins to PyPI.

## Further comments

Write here any other comment about the release, if any.

### CODE REVIEW

File: aea/cli/test.py [hunk 1]
```diff
@@ -423,33 +423,35 @@ def load_package(
...
+    else:
+        if not (aea_project_path and aea_project_path.exists()):
+            raise click.ClickException(
+                "aea_project_path not spcified or does not exists"
+            )
+
+        agent_config = AEABuilder.try_to_load_agent_configuration_file(
+            cast(Path, aea_project_path)
+        )
+        load_aea_packages_recursively(agent_config, package_path_finder, root_packages)
```
Comment (angrybayblade) [inline aea/cli/test.py:454]: This will fail when running an agent test by path, ie using `aea test by-path PATH_TO_AGENT`

Comment (solarw) [reply aea/cli/test.py]: @angrybayblade 
updated. test written

File: aea/cli/test.py [hunk 2]
```diff
@@ -423,33 +423,33 @@ def load_package(
...
-        click.ClickException,
-    )
+    else:
+        # already know it's an agent
+        aea_project_path = aea_project_path or package_dir
+
+        agent_config = AEABuilder.try_to_load_agent_configuration_file(
+            cast(Path, aea_project_path)
+        )
+        load_aea_packages_recursively(agent_config, package_path_finder, root_packages)
```
Comment (angrybayblade) [inline aea/cli/test.py:452]: Does this load `packages.AUTHOR.agents.AGENT` package? Can you try running test on the autonomy repo with an editable installation of this branch?

Comment (solarw) [reply aea/cli/test.py]: @angrybayblade 
fixed, tested over  openautonomy.

**Review by angrybayblade — APPROVED** [review]
LGTM

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**